### ISSUE TITLE
Referencing YubiKeyDeviceListener.Instance throws System.TypeInitializationException on all versions after 1.2

### ISSUE THREAD

**Issue by mjfara** [Yubico/Yubico.NET.SDK#24]
Any reference to YubiKeyDeviceListener.Instance throws an exception.
Version 1.2 is the last working version.
.NET Framework 4.8

`YubiKeyDeviceListener.Instance.Arrived += YubiKey_Arrived;`

```
System.TypeInitializationException: 'The type initializer for 'PerTypeValues`1' threw an exception.'

FileLoadException: Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

```

**Comment by GregDomzalski** [comment]
Hi @mjfara,

I tried to create a project to reproduce this, but I was not successful.

Here's what I did:
- Created a console project based on .NET Framework 4.8
- Used PackageReferences to include Yubico.YubiKey, since the SDK does not currently support packages.config
- Built as Debug and Release using x64 architecture, as 'Any CPU' will not work since we have native dependencies.
- Added event handlers for Arrived and Removed.

The app was able to run and I saw the expected arrive and remove event messages.

Would you be able to package up a minimal repro app for me to investigate further?

I also found this StackOverflow thread that seems like it might be relevant? https://stackoverflow.com/questions/62764744/could-not-load-file-or-assembly-system-runtime-compilerservices-unsafe

**Comment by mjfara** [comment]
When trying to create a minimal repro, I now get the following error when trying to add the 1.4.2 package to an empty class library:

```
Could not install package 'Yubico.NativeShims 1.3.1'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.8', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.
```

**Comment by GregDomzalski** [comment]
Yes, 1.2 is when we added device event support and refactored a lot of our interop code into a custom library. It's unfortunately cause headaches for our .NET Framework 4.x based users. We have yet to discover a good fix for this.

This error is because your new project is using `packages.config` which is not supported.  If the project already has NuGet references, there should be a right-click option on the project or the project references item in solution explorer to "migrate to PackageReferences" or something like that.

Otherwise you can unload the project file and open it in the editor and add:
```xml
  <ItemGroup>
    <PackageReference Include="Yubico.YubiKey">
      <Version>1.4.2</Version>
    </PackageReference>
  </ItemGroup>
```
somewhere before the final import of the project targets (usually the second to last line in the file).

**Comment by mjfara** [comment]
Thanks for the info.
I migrated to PackageReferences but still get the same error, here is a repro:

[YubiTest.zip](https://github.com/Yubico/Yubico.NET.SDK/files/9719204/YubiTest.zip)

**Comment by mattfara-okta** [comment]
Hi @GregDomzalski , any chance you were able to take a look at that minimal repro?

**Comment by GregDomzalski** [comment]
Yes I did. I was able to compile and run the application. Unfortunately I still was not able to reproduce this. 

I suspect that this may be a machine-specific issue. Are you able to try this out on a second machine?

The SO link I provided may be on to something: perhaps the version of this library registered in the GAC is different from what your current runtime requires. I'm much less familiar with .NET Framework and the GAC - is there a way you can confirm or force the installation of the library version is asking for?

**Comment by mjfara** [comment]
I'm also beginning to think it's machine specific issue.
I'll have to circle back to this issue at a later time and will simply avoid using the listener events for now. Thanks for looking into it.

**Comment by GregDomzalski** [comment]
Hi @mjfara - any progress on your side? Were you able to get past this issue?

Let me know if I can close this issue.

**Comment by mjfara** [comment]
Unfortunately had to manually bundle NativeShims.dll in the installer in order to get this to work properly.
Can confirm there are no problems on .NET6, but .NET Framework has provided some headaches.
